### PR TITLE
[RFC][UN-15294]add uni-calendar component

### PR DESCRIPTION
**Pull request by niceoutput**
https://uniplaces.atlassian.net/browse/UN-15294

screen shot 2018-11-30 at 18 13 29

## Show Above
screen shot 2018-11-30 at 16 34 58

## Show Below
screen shot 2018-11-30 at 18 23 02

## Mobile
screen shot 2018-11-27 at 16 18 09

### CODE REVIEW

File: addon/templates/components/uni-calendar.hbs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{{#if label}}
+  <label class="uni-datepicker-input__label">{{label}}</label>
+{{/if}}
+
+<div class="uni-calendar">
+  {{uni-datepicker-input placeholder=placeeholder showAbove=showAbove}}
```
Comment (DaniFrancisco) [inline addon/templates/components/uni-calendar.hbs]: fix typo

File: addon/templates/components/uni-calendar.hbs [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  {{uni-datepicker-input placeholder=placeeholder showAbove=showAbove}}
```
Comment (DaniFrancisco) [inline addon/templates/components/uni-calendar.hbs]: fix typo

File: app/components/uni-calendar.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from 'ember-cli-uniq/components/uni-calendar';
```
Comment (DaniFrancisco) [inline app/components/uni-calendar.js]: add new line

File: tests/dummy/app/templates/components/demo-page.hbs [hunk 4]
```diff
@@ -252,6 +252,18 @@
...
+        {{uni-title title="uni-calendar"}}
+        {{component-links component="uni-calendar"}}
+        <p class="description">
+          This component shows two datepickers one with a move-in date and the other with move-out. Both are inputs. It's also possible to open/close the datepicker
+          by clicking on the input. You can provide a placeholder for the input.
+        </p>
+        {{uni-calendar}}
+        {{code-snippet name="uni-calendar.hbs"}}
+      </div>
+
```
Comment (DaniFrancisco) [inline tests/dummy/app/templates/components/demo-page.hbs]: rm

File: tests/integration/components/uni-calendar-test.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent('uni-calendar', 'Integration | Component | uni calendar', {
+  integration: true
+});
+
+test('it renders', function(assert) {
+  // Set any properties with this.set('myProperty', 'value');
```
Comment (DaniFrancisco) [inline tests/integration/components/uni-calendar-test.js]: add proper testing

**Review by DaniFrancisco — CHANGES_REQUESTED** [review]

File: addon/styles/partials/_uni-datepicker-input.scss [hunk 6]
```diff
@@ -35,4 +42,20 @@
       }
     }
   }
+
+  .uni-datepicker {
+    position: absolute;
+    top: $input-height + $datepicker-safe-margin;
```
Comment (mmelo) [inline addon/styles/partials/_uni-datepicker-input.scss]:
```suggestion
    top: #{$input-height + $datepicker-safe-margin};
```

File: addon/styles/partials/_uni-datepicker-input.scss [hunk 7]
```diff
@@ -35,4 +42,20 @@
...
+    z-index: 10;
+    background: color("primary", "highlight");
+    box-shadow: 0 8px 16px 0 rgba(130, 141, 147, 0.35);
+    padding: 24px;
+
+    &.show-above {
+      position: absolute;
+      top: auto;
+      bottom: $input-height + $datepicker-safe-margin;
```
Comment (mmelo) [inline addon/styles/partials/_uni-datepicker-input.scss]:
```suggestion
      bottom: #{$input-height + $datepicker-safe-margin};
```

File: tests/integration/components/uni-calendar-test.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+module('Integration | Component | uni calendar', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders both inputs', async function(assert) {
+    assert.expect(1);
+
+    await render(hbs`{{uni-calendar}}`);
+
+    assert.dom('.uni-datepicker-input__wrapper').exists({ count: 2 });
+  });
```
Comment (mmelo) [inline tests/integration/components/uni-calendar-test.js:15]: can you add a test for the opening/closing behaviour?

**Review by mmelo — CHANGES_REQUESTED** [review]

**Review by DaniFrancisco — APPROVED** [review]

**Review by mmelo — APPROVED** [review]

**Pull request merged**